test/config_tests.ml

```diff
@@ -234,6 +234,29 @@ tls-auth [inline]
     (Ok expected)
     (parse_noextern with_newlines)
 
+let string_of_file filename =
+  try
+    let fh = open_in filename in
+    let content = really_input_string fh (in_channel_length fh) in
+    close_in_noerr fh ;
+    content
+  with _ -> Alcotest.failf "Error reading file %S" filename
+
+let config_dir = "sample-configuration-files/"
+
+let parse_client_configuration name () =
+  Unix.chdir config_dir;
+  Fun.protect ~finally:(fun () -> Unix.chdir "..")
```

Comment:
Hmm, `chdir` inside a test suite seems like a risky thing to be doing since we don't know what else is writing files, can't we make do with absolute paths?

Reply:
since the configuration files refer to "files on the local system", it is hard to do with absolute paths.

in my perspective, the `Fun.protect` does exactly the right thing here -- ensuring that `Unix.chdir ".."` is always executed before the test is finished.

Reply:
Oh, this raises the question how openvpn does this - whether relative paths are relative to current execution dir or the config file itself. I would expect it to be relative to the config file (in which case we shouldn't need `chdir`).

Reply:
well, we'd need the logic for the parser to do relative to the configuration file -- which atm is not present (I tried first without `chdir`, it failed, so I came up with this) ;)

but if that's something you think should be implemented by the config parser, please feel free (you could push this to this PR if you like). this would as well get rid of `Fun.protect` and the 4.08.0 lower bound.

Reply:
(but we may need to adjust the config file parser if that's the case)

Reply:
well, the original `openvpn` actually takes files referenced in a configuration as relative to the current working directory -- i.e. `openvpn config/client.conf` (with embedded `ca ca.crt`) fails:
```bash
Options error: --ca fails with 'ca.crt': No such file or directory (errno=2)
Sun Mar 29 22:33:58 2020 us=707536 WARNING: cannot stat file 'ta.key': No such file or directory (errno=2)
```

Reply:
but openvpn has the `--cd` argument to change its working directory "before reading any files such as configuration files"... not immediately useful here since we're using it as a library..

Reply:
This means that the semantics of the config file depends on the working directory, which seems really ugly to me (ie execution the same openvpn commandline with the same arguments can lead to different results based on what the cwd is).
It also implies that e.g. our tool to read/check configuration files should also work different depending on your cwd (but with the same file).
I'd be super inclined to change that to `~/.ssh/config`-style semantics where paths are relative to the location of the file that makes the reference, so a config file can only be parsed one way.

Reply:
I did so in 4a0c60f.